root/static/scripts/common/components/CDTocReleaseListRow.js

```diff
@@ -0,0 +1,191 @@
+/*
+ * @flow strict
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import ReleaseCatnoList from '../../../../components/ReleaseCatnoList.js';
+import ReleaseLabelList from '../../../../components/ReleaseLabelList.js';
+import {SanitizedCatalystContext} from '../../../../context.mjs';
+import MediumTracklist from '../../../../medium/MediumTracklist.js';
+import loopParity from '../../../../utility/loopParity.js';
+import type {ReleaseWithMediumsAndReleaseGroupT}
+  from '../../relationship-editor/types.js';
+import bracketed from '../utility/bracketed.js';
+import formatBarcode from '../utility/formatBarcode.js';
+import mediumFormatName from '../utility/mediumFormatName.js';
+
+import ArtistCreditLink from './ArtistCreditLink.js';
+import EntityLink from './EntityLink.js';
+import ReleaseEvents from './ReleaseEvents.js';
+import TaggerIcon from './TaggerIcon.js';
+
+component CDTocReleaseListRowMediums(
+  associatedMedium?: number,
+  index: number,
+  loopClass: string,
+  medium: MediumWithRecordingsT,
+) {
+  const [hidden, setHidden] = React.useState<boolean>(true);
+
+  function onButtonClick(event: SyntheticMouseEvent<HTMLAnchorElement>) {
+    event.preventDefault();
+    hidden ? setHidden(false) : setHidden(true);
+  }
+
+  const cdTocAlreadyAttached =
+  Boolean(associatedMedium) && (medium.id === associatedMedium);
+  const cannotHaveDiscIds = !medium.may_have_discids;
+  const hasLoadedTracks = Boolean(medium.tracks);
+
+  const formatAndPosition = texp.l('{medium_format} {position}', {
+    medium_format: mediumFormatName(medium),
+    position: medium.position,
+  });
+
+  const formatPositionAndName = nonEmpty(medium.name)
+    ? texp.l('{medium_format_and_position}: {medium_name}', {
+      medium_format_and_position: formatAndPosition,
+      medium_name: medium.name,
+    }) : formatAndPosition;
+
+  return (
+    <React.Fragment key={index}>
```

Comment:
This key doesn't do anything because there's no list here. Keys are only needed on the `CDTocReleaseListRowMediums` elements themselves where they're used in a list. It appears this is the only use of the `index` prop, so it can be removed if so